src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -279,58 +251,40 @@ public TElement FirstOrDefault(Func<TElement, bool> predicate)
             }
         }
 
-        public TElement Last()
+        public TElement TryGetLast(out bool found)
         {
-            CachingComparer<TElement> comparer = GetComparer();
             using (IEnumerator<TElement> e = source.GetEnumerator())
             {
-                if (!e.MoveNext()) throw Error.NoElements();
-                TElement value = e.Current;
-                comparer.SetElement(value);
-                while (e.MoveNext())
+                if (!e.MoveNext())
                 {
-                    TElement x = e.Current;
-                    if (comparer.Compare(x, false) >= 0) value = x;
+                    found = false;
+                    return default(TElement);
                 }
-                return value;
-            }
-        }
 
-        public TElement LastOrDefault()
-        {
-            CachingComparer<TElement> comparer = GetComparer();
-            using (IEnumerator<TElement> e = source.GetEnumerator())
-            {
-                if (!e.MoveNext()) return default(TElement);
+                CachingComparer<TElement> comparer = GetComparer();
                 TElement value = e.Current;
                 comparer.SetElement(value);
                 while (e.MoveNext())
                 {
-                    TElement x = e.Current;
-                    if (comparer.Compare(x, false) >= 0) value = x;
+                    TElement current = e.Current;
+                    if (comparer.Compare(current, false) >= 0) value = current;
```

Comment:
Nit: could be done in a future change, but... named arg for the Boolean?  I have no idea what ```false``` means here :smile: